### ISSUE TITLE
Iteration callback for early stopping 

### ISSUE THREAD

**Issue by gkronber** [cvxgrp/scs#227]
## Specifications
- OS: Windows
- SCS Version: 3.2.1
- Compiler: Microsoft C++

## Description
It would be nice to be able to stop SCS early. One way to implement this could be be passing a function pointer in the ScsSettings that is called on each iteration. This would also allow to report the progress back to the caller. Right now SCS writes directly to SCS and there is no way for the caller to monitor optimization progress.

**Comment by bodono** [comment]
Yes, that would be a nice feature to have. It would be relatively straightforward to add support for a callback function that takes the relevant data structs and returns a boolean for whether or not to stop, which would allow custom stopping criteria. I'm pretty swamped at the moment but I would be happy to accept a PR for this.
